tss-esapi/src/abstraction/web_authn.rs

```diff
@@ -0,0 +1,187 @@
+use crate::{
+    constants::SessionType,
+    handles::{KeyHandle, SessionHandle},
+    interface_types::algorithm::HashingAlgorithm,
+    structures::{
+        pcr_selection_list::PcrSelectionListBuilder, Attest, HashScheme, Public, Signature,
+        SignatureScheme, SymmetricDefinition,
+    },
+    traits::Marshall,
+    Context, Result,
+};
+use ciborium::cbor;
+use std::convert::TryInto;
+
+#[derive(Debug)]
+pub struct TpmStatement {
+    attestation: Attest,
+    signature: Signature,
+    public_area: Public,
+    algorithm: SignatureScheme,
+    x509_chain: Vec<Vec<u8>>,
+}
+
+impl TpmStatement {
+    pub fn new(
+        context: &mut Context,
+        object: KeyHandle,
+        key: KeyHandle,
+        nonce: Vec<u8>,
+    ) -> Result<TpmStatement> {
+        let (public_area, _, _) = context.read_public(object)?;
+        let (session_1, _, _) = context.sessions();
```

Comment:
I think you should create `session_1` in this function too, instead of pulling it from `context`, because we don't know if the context we get as input has any sessions